src/api/app/models/token/workflow.rb

```diff
@@ -69,6 +69,10 @@ def members
     [users, groups&.map(&:users)&.flatten].flatten.compact.uniq
   end
 
+  def unique_repository
+    workflow_runs.where.not(response_url: nil).first
```

Comment:
Please don't "hide" standard AR in instance methods, it hurts readability. Either use a scope or in case of single use or simple query just use standard AR directly in the code. Everybody needs to look up what `@token.unique_repository` does, no one needs to look up `workflow_runs.where` does.